music_assistant/providers/audioscrobbler/__init__.py

```diff
@@ -0,0 +1,315 @@
+"""Allows scrobbling of tracks with the help of PyLast."""
+
+import logging
+import time
+from typing import TYPE_CHECKING
+
+import pylast
+from music_assistant_models.config_entries import (
+    ConfigEntry,
+    ConfigValueOption,
+    ConfigValueType,
+    ProviderConfig,
+)
+from music_assistant_models.constants import SECURE_STRING_SUBSTITUTE
+from music_assistant_models.enums import ConfigEntryType, EventType, PlayerState
+from music_assistant_models.errors import LoginFailed, SetupFailedError
+from music_assistant_models.event import MassEvent
+from music_assistant_models.media_items import Album, is_track
+from music_assistant_models.provider import ProviderManifest
+
+if TYPE_CHECKING:
+    from music_assistant_models.media_items import Track
+    from music_assistant_models.player_queue import PlayerQueue
+
+from music_assistant import MusicAssistant
+from music_assistant.constants import MASS_LOGGER_NAME
+from music_assistant.helpers.auth import AuthenticationHelper
+from music_assistant.models import ProviderInstanceType
+from music_assistant.models.plugin import PluginProvider
+
+
+async def setup(
+    mass: MusicAssistant, manifest: ProviderManifest, config: ProviderConfig
+) -> ProviderInstanceType:
+    """Initialize provider(instance) with given configuration."""
+    provider = ScrobblerProvider(mass, manifest, config)
+    pylast.logger.setLevel(provider.logger.level)
+    if provider.logger.level == logging.DEBUG:
+        # httpcore is quite spammy without providing useful information 99% of the time
+        logging.getLogger("httpcore").setLevel(logging.INFO)
+
+    return provider
+
+
+class ScrobblerProvider(PluginProvider):
```

Comment:
Not sure if the idea is to extend this scrobble provider with other targets but otherwise I think should be renamed to LastFM Scrobble Provider and mention in the readme/description that its also suitable for LibreFM.

Reply:
this comment remains, the providers is targeted to be a lastfm/librefm scrobble provider